README.md

```diff
@@ -577,7 +598,23 @@ On the exemple above, the `result` will be:
 }
 ```
 
-## Improve type inference with Utility Types
+## Runtime utilities
+### fromSuccess
+Whenever the composition utilities fall short and you want to call other domain functions inside your current one, you can use the `fromSuccess` function to create a domain function that is expected to always succeed.
+```ts
+const domainFunctionA = makeDomainFunction(
+  z.object({ id: z.string() }),
+)(async ({ id }) => {
+  const valueB = await fromSuccess(domainFunctionB)({ userId: id })
+  // do something else
+  return {
+    valueA,
+    valueB,
+  }
+})
+```
+
+Otherwise, if the domain function applied to `fromSuccess` happens to fail, the error will be bubbled up exactly as it was thrown.
```

Comment:
same thing, the error comes out as an exception. The exception data has a similar shape to `ErrorResult` so it can handle any error that would be returned by a DF.

Reply:
The test case I added is precisely a failure on input parsing.